rapid7_insightvm/komand_rapid7_insightvm/util/resource_helpers.py

```diff
@@ -0,0 +1,182 @@
+            role_ids = [r['id'] for r in roles]
+
+            error = f"ERROR: Specified role ID ({role_id}) does not exist, valid role IDs: {role_ids}"
+            self.logger.error(error)
+            raise ResourceNotFound(error)
+
+    def validate_user_permissions(self, console_url: str, user: dict) -> dict:
+        """
+        Validate that user permissions are appropriate given the user's role.
+        This is useful when updating a user's role or create a new user as
+        some roles required the allSites, allAssetGroups, or superuser parameters
+        to be set.
+        :param console_url: URL to the InsightVM console
+        :param user: dict containing the details for a user (from the InsightVM API or matching the format)
+        :return: The user dict with updated permissions if required
+        """
+        endpoint = endpoints.Role.roles(console_url, user['role']['id'])
+        role = self.requests.resource_request(endpoint=endpoint)
+
+        # Set All Sites permission if required and not set
+        if not user['role']['allSites']:
+            for permission in ("all-permissions", "manage-sites", "manage-tags"):
+                if permission in role['privileges']:
+                    self.logger.info(
+                        f"Setting 'Access All Sites' to 'true' as it is required based "
+                        f"on the permissions for the role '{user['role']['id']}'"
+                    )
+                    user['role']['allSites'] = True
+                    break
+
+        # Set All Asset Groups permission if required and not set
+        if not user['role']['allAssetGroups']:
+            for permission in ("all-permissions", "manage-dynamic-asset-groups"):
+                if permission in role['privileges']:
+                    self.logger.info(
+                        f"Setting 'Access All Asset Groups' to 'true' as it is required based "
+                        f"on the permissions for the role '{user['role']['id']}'"
+                    )
+                    user['role']['allAssetGroups'] = True
+                    break
+
+        # Set superuser permission if required and not set
+        # Do this silently as it isn't exposed to the user of the plugin and
+        # appears to simply be an internal setting that custom roles cannot have
+
+        if not user['role']['superuser'] and (user['role']['id'] == 'global-admin'):
+            user['role']['superuser'] = True
+
+        return user
+
+    def validate_user_email(self, email: str) -> None:
+        """
+        Validate a user email address string.
+        :param email: Email address to validate
+        :return: None, raises an error if validation fails
+        """
+        email_regex = re.compile(r'[^@]+@[^@]+\.[^@]+')
+
+        if not email_regex.match(email):
+            error = 'ERROR: Email address for user account was not valid!'
+            self.logger.error(error)
+            raise Exception(error)
```

Comment:
We should be using PluginException - and cut out ERROR since SDK already prefixes line with a failure message